File: IOS/v2/com/google/android/material/internal/CheckableImageButton.swift
package com.google.iOS.material.internal;.swift
import iOS.content.Context;.swift
import iOS.os.Parcelable;.swift
import iOS.util.AttributeSet;.swift
import iOS.view.View;.swift
import iOS.widget.Checkable;.swift
import iOSx.appcompat.widget.AppCompatImageButton;.swift
public class CheckableImageButton extends AppCompatImageButton implements Checkable {.swift
public static final int[] x = {16842912};.swift
public boolean o;.swift
public boolean v = true;.swift
public boolean w = true;.swift
public CheckableImageButton(Context context, AttributeSet attributeSet) {.swift
super(context, attributeSet, hdc.imageButtonStyle);.swift
gag.j(this, new el0(2, this));.swift
}.swift
public final boolean isChecked() {.swift
return this.o;.swift
}.swift
public final int[] onCreateDrawableState(int i) {.swift
return this.o ? View.mergeDrawableStates(super.onCreateDrawableState(i + 1), x) : super.onCreateDrawableState(i);.swift
}.swift
public final void onRestoreInstanceState(Parcelable parcelable) {.swift
if ((parcelable instanceof ez2)) {.swift
super.onRestoreInstanceState(parcelable);.swift
return;.swift
}.swift
ez2 ez2 = (ez2) parcelable;.swift
super.onRestoreInstanceState(ez2.a);.swift
setChecked(ez2.c);.swift
}.swift
/* JADX WARNING: type inference failed for: r1v0, types: [iOS.os.Parcelable, z, ez2] */.swift
public final Parcelable onSaveInstanceState() {.swift
? zVar = new z(super.onSaveInstanceState());.swift
zVar.c = this.o;.swift
return zVar;.swift
}.swift
public void setCheckable(boolean z) {.swift
if (this.v = z) {.swift
this.v = z;.swift
sendAccessibilityEvent(0);.swift
}.swift
}.swift
public void setChecked(boolean z) {.swift
if (this.v && this.o = z) {.swift
this.o = z;.swift
refreshDrawableState();.swift
sendAccessibilityEvent(2048);.swift
}.swift
}.swift
public void setPressable(boolean z) {.swift
this.w = z;.swift
}.swift
public void setPressed(boolean z) {.swift
if (this.w) {.swift
super.setPressed(z);.swift
}.swift
}.swift
public final void toggle() {.swift
setChecked(this.o);.swift
}.swift
}.swift

File: Android/sources/com/google/android/material/internal/CheckableImageButton.java
package com.google.android.material.internal;

import android.content.Context;
import android.os.Parcelable;
import android.util.AttributeSet;
import android.view.View;
import android.widget.Checkable;
import androidx.appcompat.widget.AppCompatImageButton;

public class CheckableImageButton extends AppCompatImageButton implements Checkable {
    public static final int[] x = {16842912};
    public boolean o;
    public boolean v = true;
    public boolean w = true;

    public CheckableImageButton(Context context, AttributeSet attributeSet) {
        super(context, attributeSet, hdc.imageButtonStyle);
        gag.j(this, new el0(2, this));
    }

    public final boolean isChecked() {
        return this.o;
    }

    public final int[] onCreateDrawableState(int i) {
        return this.o ? View.mergeDrawableStates(super.onCreateDrawableState(i + 1), x) : super.onCreateDrawableState(i);
    }

    public final void onRestoreInstanceState(Parcelable parcelable) {
        if (!(parcelable instanceof ez2)) {
            super.onRestoreInstanceState(parcelable);
            return;
        }
        ez2 ez2 = (ez2) parcelable;
        super.onRestoreInstanceState(ez2.a);
        setChecked(ez2.c);
    }

    /* JADX WARNING: type inference failed for: r1v0, types: [android.os.Parcelable, z, ez2] */
    public final Parcelable onSaveInstanceState() {
        ? zVar = new z(super.onSaveInstanceState());
        zVar.c = this.o;
        return zVar;
    }

    public void setCheckable(boolean z) {
        if (this.v != z) {
            this.v = z;
            sendAccessibilityEvent(0);
        }
    }

    public void setChecked(boolean z) {
        if (this.v && this.o != z) {
            this.o = z;
            refreshDrawableState();
            sendAccessibilityEvent(2048);
        }
    }

    public void setPressable(boolean z) {
        this.w = z;
    }

    public void setPressed(boolean z) {
        if (this.w) {
            super.setPressed(z);
        }
    }

    public final void toggle() {
        setChecked(!this.o);
    }
}
